bufferflow_timedbinary.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"encoding/json"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type BufferflowTimedBinary struct {
+	Name   string
+	Port   string
+	Output chan []byte
+	Input  chan []byte
+	done   chan bool
+	ticker *time.Ticker
+}
+
+var (
+	bufferedOutputBinary []byte
+	sPortBinary          string
+)
+
+func (b *BufferflowTimedBinary) Init() {
+	log.Println("Initting timed buffer binary flow (output once every 16ms)")
+	bufferedOutputBinary = nil
+	sPortBinary = ""
+
+	go func() {
+		b.ticker = time.NewTicker(16 * time.Millisecond)
+		b.done = make(chan bool)
+	Loop:
+		for {
+			select {
+			case data := <-b.Input:
+				bufferedOutputBinary = append(bufferedOutputBinary, data...)
+				sPortBinary = b.Port
+			case <-b.ticker.C:
+				if bufferedOutputBinary != nil {
+					m := SpPortMessageRaw{sPortBinary, bufferedOutputBinary}
+					buf, _ := json.Marshal(m)
+					b.Output <- buf
+					bufferedOutputBinary = nil
+					sPortBinary = ""
+				}
+			case <-b.done:
+				break Loop
+			}
+		}
+
+		close(b.Input)
```

Comment:
Remove this, `BufferflowTimedBinary.Close()` does it.

Reply:
Am talking about this `Close()` that is in this same file, which one are you talking about? 🤔 
```
func (b *BufferflowTimedBinary) Close() {
	b.ticker.Stop()
	close(b.Input)
}
```

Reply:
oh! That's a bug! :bug: It should be:
```
func (b *BufferflowTimedBinary) Close() {
	b.ticker.Stop()
	b.done <- true
	close(b.done)
}
```